core/rend/gles/gles.cpp

```diff
@@ -1964,61 +1979,81 @@ bool RenderFrame()
 
 		ShaderUniforms.Set(s);
 	}
+
 	//setup render target first
 	if (is_rtt)
 	{
-		GLuint channels,format;
+		const char * renderer = (char *)glGetString(GL_RENDERER);
+		const char * adreno506Renderer = "Adreno (TM) 506";
+		GLuint channels, format;
 		switch(FB_W_CTRL.fb_packmode)
 		{
 		case 0: //0x0   0555 KRGB 16 bit  (default)	Bit 15 is the value of fb_kval[7].
 			channels=GL_RGBA;
-			format=GL_UNSIGNED_SHORT_5_5_5_1;
+			// When using RGBA5551 format on Adreno 506, rendering is extremely slow
```

Comment:
Keep in mind, powervr will always do 32bpp rendering. The frame-buffer format only when copying to memory from the internal tile storage.

Reply:
Ok, but at the end it will be 16bpp anyway for RTT? This solution uses 16bpp rendering from the beginning because it should be faster in most cases (for slower mobile GPUs) than 32bpp rendering, should something be changed?